sdk/cosmos/src/errors.rs

```diff
@@ -1,45 +1,51 @@
-/// A general error having to do with Cosmos.
-#[allow(missing_docs)]
+/// An error having to do with Cosmos.
 #[non_exhaustive]
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
```

Comment:
For the purposes of establishing guidelines, is the eventual goal that every crate has a single `Error` type e.g. azure_core::Error, azure_cosmos::Error, etc.? I'm working on cleaning up azure_identity, for example, and it has numerous error types strewn throughout the crate.

Reply:
Exactly - there will be one "main" error for each crate but there might be some other error types for specific situations. Let's say for instance only one service needs to parse a specific data format (let's call the service `Foo` and the data format `FooData`). Then the `Foo` SDK would probably have an error type `foo::Error` with a variant that contains another error type for specific errors about parsing `FooData` (i.e., `FooDataParsingError(FooDataParsingError)`)